blockchain/notifications.go

```diff
@@ -21,10 +21,31 @@ type NotificationCallback func(*Notification)
 
 // Constants for the type of a notification message.
 const (
+	// NTNewTipBlockChecked indicates the associated block intends to extend
+	// the current main chain and has passed all of the sanity and
+	// contextual checks such as having valid proof of work, valid merkle
+	// and stake roots, and only containing allowed votes and revocations.
+	//
+	// It should be noted that the block might still ultimately fail to
+	// become the new main chain tip if it contains invalid scripts, double
+	// spends, etc.  However, this is quite rare in practice because a lot
+	// of work was expended to create a block which satisifies the proof of
+	// work requirement.
+	//
+	// Finally, this notification is only sent if the the chain is believed
+	// to be current and the chain lock is NOT released, so consumers must
+	// take care to avoid calling blockchain functions to avoid potential
+	// deadlock.
+	//
+	// Typically, a caller would want to use this notification to relay the
+	// block to the rest of the network without needing to wait for the more
+	// time consuming full connection to take place.
+	NTNewTipBlockChecked NotificationType = iota
```

Comment:
why can't this be added to the end, to leave the other constants the same values?

Reply:
I prefer to have them defined in the order they are invoked.

Reply:
But it's a breaking change